### PR TITLE
Adds support for GCM Unicast

### PR DESCRIPTION
**Pull request by howleysv**
Updates library to current API references provided by google:

https://developers.google.com/cloud-messaging/http
https://developers.google.com/cloud-messaging/http-server-ref

### CODE REVIEW

File: lib/gcm.ex [hunk 1]
```diff
@@ -28,10 +30,15 @@ defmodule GCM do
...
         invalid_registration_ids: [], not_registered_ids: [], status_code: 200,
         success: 2}}
+  ```
   """
-  @spec push(binary, [binary], map | [Keyword]) :: { :ok, map } | { :error, term }
+  @spec push(String.t, String.t | [String.t], Map.t | Keyword.t) :: { :ok, Map.t } | { :error, term }
   def push(api_key, registration_ids, options \\ %{}) do
-    body = %{ registration_ids: registration_ids }
+    registration_ids = List.wrap(registration_ids)
+    body = case registration_ids do
```
Comment (edgurgel) [inline lib/gcm.ex:38]: Why not just 

```
case registration_ids do
  ids when is_list(ids) -> %{ registration_ids: ids }
  id -> %{ to: id } 
end
```

So you don't need to `List.wrap`

**Comment by edgurgel** [comment]
What's the difference between sending `to` and `registration_ids`? Is there any difference on the response?

File: lib/gcm.ex (hunk 1 above)
Comment (howleysv) [inline lib/gcm.ex:38]: I also want to use `:to` when handling the case of a single element list

**Comment by howleysv** [comment]
Right now it seems to be the same, but I guess it could change since it's outside the current spec. There is a note on the  `registration_ids` parameter:

> Use this parameter only for multicast messaging, not for single recipients.

**Comment by edgurgel** [comment]
Seems legit. I will use it with a real app just to "smoke test" and I will merge. Thanks for the PR!

**Comment by howleysv** [comment]
Sounds good :+1: 

FWIW We're using this code in production :shipit:

**Comment by edgurgel** [comment]
Awesome! I will release a new version soon! :+1:

**Comment by edgurgel** [comment]
Released https://hex.pm/packages/gcm/1.2.0 :tada: Thank you!

**Pull request merged**